Pinta.Core/ImageFormats/GdkPixbufFormat.cs

```diff
@@ -38,46 +38,39 @@ public GdkPixbufFormat (string filetype)
 		this.filetype = filetype;
 	}
 
-	#region IImageImporter implementation
-
-	public void Import (Gio.File file, Gtk.Window parent)
+	public Document Import (Gio.File file, Gtk.Window parent)
 	{
-		Pixbuf bg;
+		Pixbuf streamBuffer = ReadPixbuf (file);
+		Pixbuf effectiveBuffer = streamBuffer.ApplyEmbeddedOrientation () ?? streamBuffer;
 
-		// Handle any EXIF orientation flags
-		using (var fs = file.Read (cancellable: null)) {
-			try {
-				bg = Pixbuf.NewFromStream (fs, cancellable: null)!; // NRT: only nullable when an error is thrown
-			} finally {
-				fs.Close (null);
-			}
-		}
+		Size imageSize = new (effectiveBuffer.Width, effectiveBuffer.Height);
 
-		bg = bg.ApplyEmbeddedOrientation () ?? bg;
+		Document newDocument = new (imageSize, file, filetype);
+		newDocument.Workspace.ViewSize = imageSize;
```

Comment:
Should the `Document` constructor also update the `ViewSize`, since all the callers are doing this manually anyways?

Reply:
Good catch. I'm now updating it from within `Document`'s constructor